p2p/libp2p/netMessenger.go

```diff
@@ -493,6 +500,22 @@ func (netMes *networkMessenger) createConnectionMonitor(p2pConfig config.P2PConf
 	return nil
 }
 
+// PeerConnected can be called whenever a new peer is connected to this host
+func (netMes *networkMessenger) PeerConnected(pid core.PeerID) {
```

Comment:
we should think of a way to somehow close the network messenger after the bootstrapping is done. 
Why? Because we should correctly setup the network messenger with the "final" processing components able to be notified and notify as well the new peers that will interact with. Will think of a solution to this.